.github/workflows/build-and-test.yml

```diff
@@ -40,6 +40,20 @@ jobs:
     - name: Checkout Repository
       uses: actions/checkout@v4
 
+    - name: Setup Java
```

Comment:
Commit message:
- "it now" -> "it is now" or "it now is"
- maybe change "ui" to "UI"